Bottom: 7a08fa4b1bd515cd5c79b8ba955d925b7c103065
Top:    bd9239ad86767c4f5034411393f911ca7a5625e2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-02 19:42:47 +0100

Workaround crash where contour is passed as NULL to poly_PreContour



---

diff --git a/src/polygon.c b/src/polygon.c
index c33c735..54f8f9f 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
